src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -284,10 +284,14 @@ void NUMASupportInitialize()
         return;
     }
 
-    g_numaHandle = dlopen("libnuma.so", RTLD_LAZY);
+    g_numaHandle = dlopen("libnuma.so.1.0.0", RTLD_LAZY);
```

Comment:
I think `libnuma.so.1.0.0` could be left out. There is no need to probe with a specific minor and patch version.

Reply:
Adding that was actually the point of this PR. @am11 made this change to make .NET work on a system when there is no symlink libnuma.so.1, just the libnuma.so.1.0.0

Reply:
@am11 when I check nothing seems missing on Alpine:

```$ podman run -ti alpine /bin/sh -c 'apk update && apk add numactl && ls -lah /usr/lib/libnuma*'
fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/community/x86_64/APKINDEX.tar.gz
v3.13.1-98-g655b7d332a [https://dl-cdn.alpinelinux.org/alpine/v3.13/main]
v3.13.1-99-ge77be6b709 [https://dl-cdn.alpinelinux.org/alpine/v3.13/community]
OK: 13878 distinct packages available
(1/1) Installing numactl (2.0.14-r0)
OK: 6 MiB in 15 packages
lrwxrwxrwx    1 root     root          16 Feb 11 06:30 /usr/lib/libnuma.so.1 -> libnuma.so.1.0.0
-rwxr-xr-x    1 root     root       46.2K Dec 14 14:11 /usr/lib/libnuma.so.1.0.0
```

Reply:
Hmm, that's what I have also found and shown in the description above, or maybe I missed something?

SO version 1.0.0 (as opposed to the package version 2.0.14) of libnuma hasn't changed since the beginning (~10y), so it doesn't affect anything especially with the existing probing paths intact.

This PR is a minor improvement (on top of https://github.com/dotnet/coreclr/pull/12556) mainly for the sandbox environments, where we don't have/want to carry around a symlink in addition to the binary.

Reply:
`libnuma.so.1` is present on all Linux systems after installing the appropriate package.
I think we should leave out the `libnuma.so.1.0.0`: there is no need for probing for a specific minor.patch version.
Does that make sense?

Reply:
Nope. Adding the additional probe was the whole purpose of this PR for the reasons described above. :)

Reply:
@am11, I think we can swap the order of libnuma.so.1 and libnuma.so.1.0.0 here. That would address the concern that the library version might change in the future on some systems and it would still work for your case. Does it sound reasonable?

Reply:
@am11 from what is said, I don't understand what platform is missing `libnuma.so.1` and therefore requires the specific `libnuma.so.1.0.0` probe.

The version numbers on these `.so` files are "semantic versioning". When the major increases that indicates a breaking change. We're using the v1 API, so `libnuma.so.1` is the preferable file to use.